src/span.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+import * as Constants from './constants';
+import TraceContext from './trace_context';
+
+/**
+ *
+ */
+export default class Span {
+
+    /**
+     * [constructor description]
+     * @return {[type]} [description]
+     */
+    constructor(imp) {
+        this._imp = imp;
+    }
+
+    /**
+     *
+     */
+    imp() {
+        return this._imp;
+    }
+
+    // ---------------------------------------------------------------------- //
+    //
+    // ---------------------------------------------------------------------- //
+
+    traceContext() {
+        if (Constants.API_CONFORMANCE_CHECKS) {
```

Comment:
The intention is that these are zero-overhead in production. `API_CONFORMANCE_CHECKS` will be `false` in production -- and since there is a compile step, this will be dead code (not an `if (false) {`) check.  For simplicity sake, I like the idea that the conformance checks are always part of the core debug build.

(And for the record, this isn't the reality yet -- still a tiny bit of work to do in the `webpack.config.js` to do that.)